Game.h

```diff
@@ -2,18 +2,17 @@
 #define GAME_HEADER
 
 #include "Unit.h"
+#include "Player.h"
 
 class Game
 {
 public:
-    Game(const Unit&, const Unit&);
+    void Combat(Unit&, Unit&);
+    void Combat(Player&, Player&);
 
-private:
-    Unit player;
-    Unit enemy;
```

Comment:
Ha már nincsenek tárolva a harcosok, akkor ennek az osztálynak így nem sok értelme van, a metódusok akár statikusak is lehetnek. Ez így inkább csak egy namespace. Ami nem baj, de akkor ez látszódjon az interfészből: legyenek statkusak azok a függvények, vagy csináljatok egy namespace-t az osztályból.

Reply:
De ha referenciákat (`Unit&`) tárolna, az is jó megoldás lenne, elvégre ez a polimorfizmus lényege. Most ez az osztály nem használja ki, hogy a Player is Unit.